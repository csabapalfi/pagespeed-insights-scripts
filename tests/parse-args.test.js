const {check, parseArgs} = require('../lib/parse-args');

describe('parse-args', () => {
  const url = 'https://www.google.com';
  const mockArgv = (args = []) => ['node', 'pagespeed-score', ...args];

  describe('check', () => {
    it('throws if URL missing', () => {
      expect(() => check({_:mockArgv([])}))
        .toThrow('URL is required');
    });

    it('throws if URL is invalid', () => {
      expect(() => check({_:mockArgv(['oops'])}))
        .toThrow('Invalid URL');
    });

    it('allows if valid URL', () => {
      expect(check({_:mockArgv([url])}))
        .toBeTruthy()
    });

    it('throws if --save-assets used without --local', () => {
      expect(() => check({_:mockArgv([url]), saveAssets: true}))
        .toThrow('--save-assets only works with --local');
    });

    it('throws if --cpu-slowdown set without --local', () => {
      expect(() => check({_:mockArgv([url]), cpuSlowdown: 6}))
        .toThrow('--cpu-slowdown only works with --local');
    });

  });

  describe('parseArgs', () => {
    const argv = mockArgv([url]);
    it('returns defaults based on argv', () => {
      const args = parseArgs({argv})
      expect(args).toMatchSnapshot();
    });
  });
});