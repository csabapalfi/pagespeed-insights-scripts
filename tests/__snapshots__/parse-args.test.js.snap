// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parse-args parseArgs returns defaults based on argv 1`] = `
Object {
  "local": Object {
    "cpuSlowDown": undefined,
    "enabled": false,
    "saveAssets": false,
  },
  "runs": 1,
  "saveResults": false,
  "strategy": "mobile",
  "url": "https://www.google.com",
}
`;
